style/js/README.md

```diff
@@ -1,3 +1,190 @@
+JavaScript Style Guides
+=====
+
+Guides for programing JavaScript in style
+
+General
+---
+* Use [JSHint](http://jshint.com/)
+* Always use 'use strict'
+
+Parens and Braces
+---
+* Leave one space between 'if' and '('.
+
+```javascript
+if (something) {
+  //some work here;
+}
+```
+
+* Prefer this form for multiline if statments
+
+```javascript
+if (something) {
+  // do stuff
+  // even more stuff
+} else {
+  // some other stuff
+  // real stuff
+}
+```
+
+* On if statments, whenever it makes sense, you can omit the '{}'
+
+```javascript
+if (something)
+  return stuff;
+else
+  return otherStuff;
+```
+
+* Don't include spaces between 'function' and '('.
+
+```javascript
+function bar() {
+  return  foo;
+}
+
+var bar = function() {
+  return foo;
+}
+```
+
+Assignments and Declarations
+---
+* Initialize arrays with '[ ]'
+
+```javascript
+var array = [];
+```
+
+* Initialize objects with '{ }'
+
+```javascript
+var object = {};
+```
+
+* Leave one space after '{' and before '}' when initializing objects on the same line
+
+```javascript
+var map = { ready: 9, when: 4, "you are": 15 };
+```
+
+* Prefer this form for multiline object/array initialization
+
+```javascript
+var map = {
+  when: 4,
+  ready: 9,
+  "you are": 15
+};
+
+var map = [
+  1,
+  2,
+  3
+]
+```
+
+Constructors
+---
+
+```javascript
+function Foo(bar) {
+  this.bar = bar;
+}
+
+Foo.prototype.getBar = function() {
+  return this.bar;
+};
+
+var foo = new Foo("bar");
+
+// or
```

Comment:
@ronaldofs that doesn't mean it shouldn't be used. It might have it's use cases, as long as you know what you're doing and the consequences. Of course, you probably do want to share the methods most of the times (i.e., use the first option), but saying we should never use the other is an exaggeration.

But getting back on topic:

> These two produce different results, so we need to clarify that and write them down

I don't agree with this at all. These guides are not a Javascript tutorial. They're not about explaining how different things work. If these two have different results, they should be used accordingly. Do we have any particular rules or restrictions about where to use either one? If so, **that** should be documented.

Right now, this has no explanation at all, it just says `Constructors` followed by an enumeration of both of them.
That's clarifies nothing about their differences